src/figuresClasses.ts

```diff
@@ -1,11 +1,89 @@
-export interface Figure {}
+type Shape = `triangle` | `circle` | `rectangle`;
+type Color = `red` | `green` | `blue`;
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides: number[] = [this.a, this.b, this.c];
```

Comment:
[CODE STYLE] - Since the values for `a`, `b`, and `c` are already available as object fields, there's no need to create a separate `sides` array. You can directly use `this.a`, `this.b`, and `this.c` in your calculations.